app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt

```diff
@@ -123,6 +122,31 @@ class NavigationOverlayRobot {
             return SettingsRobot.Transition()
         }
 
+        /*
+         * Navigate to the settings button using keypresses and open, and maintain focus.
+         * Using click() to select buttons removes focus, so this is an alternative way to open
+         * Settings.
+         */
+        fun navigateToSettingsAndOpen(interact: SettingsRobot.() -> Unit): SettingsRobot.Transition {
+            // We hard-code this navigiation pattern because making a generic way to linearly navigate
+            // is very difficult within Espresso. Espresso does not allow branching, and relies on
+            // assertions, so a generic keypress navigation with conditional checks is not worth writing
```

Comment:
I find `Espresso does not allow branching, and relies on assertions` to be a little bit unclear.  Maybe `Espresso supports asserting view state, but not querying it.  Because of this, we can't write conditional logic based on the currently focused view.`

It's pretty verbose, but I think it communicates the problem better.